rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -674,35 +671,12 @@ private void markTxAsProcessed(BtcTransaction btcTx) throws IOException {
         provider.setHeightBtcTxhashAlreadyProcessed(btcTx.getHash(false), rskExecutionBlock.getNumber());
     }
 
-    protected void processRelease(BtcTransaction btcTx, Sha256Hash btcTxHash) throws IOException {
-        logger.debug("[processRelease] This is a release tx {}", btcTx);
-            // do-nothing
-            // We could call removeUsedUTXOs(btcTx) here, but we decided to not do that.
-            // Used utxos should had been removed when we created the release tx.
-            // Invoking removeUsedUTXOs() here would make "some" sense in theses scenarios:
-            // a) In testnet, devnet or local: we restart the RSK blockchain without changing the federation address. We don't want to have utxos that were already spent.
-            // Open problem: TxA spends TxB. registerBtcTransaction() for TxB is called, it spends a utxo the bridge is not yet aware of,
-            // so nothing is removed. Then registerBtcTransaction() for TxA and the "already spent" utxo is added as it was not spent.
-            // When is not guaranteed to be called in the chronological order, so a Federator can inform
-            // b) In prod: Federator created a tx manually or the federation was compromised and some utxos were spent. Better not try to spend them.
-            // Open problem: For performance removeUsedUTXOs() just removes 1 utxo
-
+    protected void processPegoutOrMigration(BtcTransaction btcTx, Sha256Hash btcTxHash) throws IOException {
+        String pegoutType = btcTx.getWalletOutputs(getActiveFederationWallet(false)).size() == btcTx.getOutputs().size() ? "migration" : "pegout";
+        logger.debug("[processPegoutOrMigration] This is a {} tx {}", pegoutType, btcTx);
```

Comment:
A Better Alternative For This Would Be To Check If The Retiring Federation Address Is In The Input And The Active Federation Address In The Output, Then That Is Certainly a Migration

Reply:
Why do you think that would be a better alternative? At this point, when `processPegoutOrMigration` method is called we already know the tx is a peg-out, the only thing we care about is checking if all outputs go to the same address, so it is a migration. No need to check the sender again, it is already validated.